strapi/strapi
# Fetch graphql endpoint path from graphql plugin

### What does it do?

Changes Strapi `body` and `security` middlewares to use actual graphql endpoint as configured in `graphql plugin` instead of hardcoding it.

### Why is it needed?
Strapi users are allowed to configure graphql plugin endpoint in the `config/plugins.js` file.
Without the change in this PR, updating the endpoint breaks the GraphQL Playground because the body and security middlewares rely on hardcoded value.

### How to test it?

- Have a strapi project with GraphQL playground enabled.
- Modify GraphQL plugin configuration in `config/plugins.js` to use custom endpoint value.
E.g.:

```diff
module.exports = ({ env }) => ({
  graphql: {
    config: {
-    endpoint: "/graphql",
+    endpoint: "/lqhparg",
      shadowCRUD: true,
    },
  },
});
```

- Try to access the playground URL and observe that the loading animation stays forever and browser JS console prints errors that the security settings blocked scripts from loading.

### Related issue(s)/PR(s)
- No issue has been created for this.

### Test runs
- Tried to run the test suits as suggested by the contributors guidelines.
Tests fail but they fail on the baseline too. No new test breakages have been introduced, though.
